modules/component/src/push/push.pipe.ts

```diff
@@ -67,63 +54,46 @@ import {
  */
 @Pipe({ name: 'ngrxPush', pure: false })
 export class PushPipe<S> implements PipeTransform, OnDestroy {
-  private renderedValue: S | null | undefined;
-
-  private readonly configSubject = new Subject<PushPipeConfig>();
-  private readonly config$ = this.configSubject
-    .asObservable()
-    .pipe(distinctUntilChanged());
+  private renderedValue: any | null | undefined;
 
   private readonly subscription: Unsubscribable;
   private readonly cdAware: CdAware<S | null | undefined>;
-  private readonly updateViewContextObserver: PartialObserver<
+  private readonly resetContextObserver: NextObserver<unknown> = {
+    next: (value: unknown) => (this.renderedValue = undefined),
+  };
+  private readonly updateViewContextObserver: NextObserver<
     S | null | undefined
   > = {
     // assign value that will get returned from the transform function on the next change detection
     next: (value: S | null | undefined) => (this.renderedValue = value),
   };
-  private readonly resetContextObserver: NextObserver<unknown> = {
-    next: (value: unknown) => (this.renderedValue = undefined),
-  };
-  private readonly configurableBehaviour = <T>(
-    o$: Observable<Observable<T>>
-  ): Observable<Observable<T>> =>
-    o$.pipe(
-      withLatestFrom(this.config$),
-      map(([value$, config]) => {
-        // As discussed with Brandon we keep it here
-        // because in the beta we implement configuration behavior here
-        return value$.pipe();
-      })
-    );
 
   constructor(cdRef: ChangeDetectorRef, ngZone: NgZone) {
     this.cdAware = createCdAware<S>({
-      work: setUpWork({
+      strategies: getStrategies<S>({
+        component: (cdRef as any).context,
         ngZone,
         cdRef,
-        context: (cdRef as EmbeddedViewRef<Type<any>>).context,
       }),
       updateViewContextObserver: this.updateViewContextObserver,
       resetContextObserver: this.resetContextObserver,
-      configurableBehaviour: this.configurableBehaviour,
     });
     this.subscription = this.cdAware.subscribe();
   }
 
-  transform(potentialObservable: null, config?: PushPipeConfig): null;
-  transform(potentialObservable: undefined, config?: PushPipeConfig): undefined;
-  transform(
-    potentialObservable: Observable<S> | Promise<S>,
-    config?: PushPipeConfig
-  ): S;
-  transform(
-    potentialObservable: Observable<S> | Promise<S> | null | undefined,
-    config: PushPipeConfig = { optimized: true }
-  ): S | null | undefined {
-    this.configSubject.next(config);
-    this.cdAware.next(potentialObservable);
-    return this.renderedValue;
+  transform<T>(potentialObservable: null, config?: string): null;
+  transform<T>(potentialObservable: undefined, config?: string): undefined;
+  transform<T>(
+    potentialObservable: Observable<T> | Promise<T>,
+    config?: string
+  ): T;
+  transform<T>(
+    potentialObservable: Observable<T> | Promise<T> | null | undefined,
```

Comment:
Use `ObservableInput<T> | null | undefined` here.  `ObservableInput<T>` covers the gamut of what RxJS can handle.

Reply:
Very nice!

Reply:
Done